### ISSUE TITLE
Logic error in `below()` etc. after updating `order_with_respect_to` fields

### ISSUE THREAD

**Issue by erikcw** [django-ordered-model/django-ordered-model#92]
I was just testing the `below()` method and I believe I've stumbled upon a logic error.

https://github.com/bfirsh/django-ordered-model/blob/master/ordered_model/models.py#L189

``` python
        if getattr(self, self.order_field_name) == getattr(ref, self.order_field_name):
            return
```

If the value of the order field on the object to be moved is equal to the value for the object you are trying to move below, then it should be decremented just like in the `>` case.  Keeping them both at the same value breaks sorting.

That `==` equality check should be switched to a less than `<` check.  ie "If I'm already below the other object, then there is no work to do, quit".

**Comment by erikcw** [comment]
This appears to also be the case with the `above()` method.

**Comment by emontellese** [comment]
I actually also just ran into this - I think that you are right.

**Comment by emontellese** [comment]
Actually, i think that 'return' may not be right regardless.  

If '>'  work needs to be done
If '<'  work *may* need to be done.  (i.e. just becuase it's already below doesn't mean it's in the right place below).
if '==' work needs to be done

This is just from a quick read-over though.

**Comment by emontellese** [comment]
Also it looks like the 'to()' function has some issues in some cases.    I'll put together a pull request.

**Comment by emontellese** [comment]
Please see #93 for a proposed solution to this and a related issue that I was having in the case where an ordered item is dragged from one list to another.

In this case, I am first updating the order_with_respect_to fields and then calling "above()" -- but this does not work in the trunk because the 'order' value of the dragged item is identical to the 'order' of an item in the existing list.

**Comment by shuckc** [comment]
My take on this is that we need to fix `.save()` when changing fields contained in `order_with_respect_to` to maintain the predicate  that the items are in order within the old and new groups. Attempting repair work in `.to()`, `.above()`, `.below()` methods is not a good solution. So really this ticket is the same issue as #121 and #85. I've hit this myself so how to sort it out for the first 2.0 release.

**Comment by shuckc** [comment]
Resolved by #248
